Mathlib/NumberTheory/Padics/PadicVal.lean

```diff
@@ -431,6 +431,39 @@ theorem min_le_padicValRat_add {q r : ℚ} (hqr : q + r ≠ 0) :
   (fun h => by rw [min_eq_right h, add_comm]; exact le_padicValRat_add_of_le (by rwa [add_comm]) h)
 #align padic_val_rat.min_le_padic_val_rat_add padicValRat.min_le_padicValRat_add
 
+/-- Ultrametric property of a p-adic valuation. -/
+lemma min_eq_padicValRat_add {q r : ℚ} (hqr : q + r ≠ 0) (hq : q ≠ 0) (hr : r ≠ 0)
+    (hval : padicValRat p q ≠ padicValRat p r) :
+    padicValRat p (q + r) = min (padicValRat p q) (padicValRat p r) := by
+  have Hmin := padicValRat.min_le_padicValRat_add (p := p) (hp := hp) hqr
+  wlog h : padicValRat p q < padicValRat p r generalizing q r with Hgen
+  · push_neg at h; rw [add_comm, min_comm]
+    exact Hgen (by rwa [add_comm]) hr hq hval.symm
+      (by rwa [min_comm, add_comm])
+      (hval.symm.lt_of_le h)
+  · rw [min_eq_left (le_of_lt h)] at Hmin ⊢
+    suffices Hreq : padicValRat p q ≥ padicValRat p (q + r) by linarith
+    suffices Haux : padicValRat p q ≥ min (padicValRat p (q + r)) (padicValRat p r) by
+      rw [min_def] at Haux
+      split_ifs at Haux with Hspl
+      · assumption
+      · linarith
+    calc padicValRat p q = padicValRat p ((q + r) - r) := by congr; simp
+    _ ≥ min (padicValRat p (q + r)) (padicValRat p (-r)) :=
+    ge_iff_le.mp <| le_trans (padicValRat.min_le_padicValRat_add (q := q+r) (r := -r) (by simpa))
+      (by rw [add_neg_cancel_right, add_sub_cancel])
+    _ = min (padicValRat p (q + r)) (padicValRat p r) := by rw [padicValRat.neg]
+
+lemma add_eq_of_lt {q r : ℚ} (hqr : q + r ≠ 0)
+(hq : q ≠ 0) (hr : r ≠ 0) (hval : padicValRat p q < padicValRat p r) :
+padicValRat p (q + r) = padicValRat p q :=
+by rw [min_eq_padicValRat_add hqr hq hr (ne_of_lt hval),min_eq_left (le_of_lt hval)]
```

Comment:
```suggestion
    (hq : q ≠ 0) (hr : r ≠ 0) (hval : padicValRat p q < padicValRat p r) :
    padicValRat p (q + r) = padicValRat p q := by
  rw [min_eq_padicValRat_add hqr hq hr (ne_of_lt hval),min_eq_left (le_of_lt hval)]
```

Reply:
Try and follow this indentation style throughout the file (look at files in mathlib for more examples) but we do 4 spaces for the statement and 2 for the proof now (and tend to put by on the previous line)